.circleci/config.yml

```diff
@@ -7,62 +7,76 @@ version: 2.1
+          root: /home/circleci/go
+          # Must be relative path from root
           paths:
-            - /go
-            - /usr/local/bin/golangci-lint
-            - /usr/local/kubebuilder
-            - /usr/local/bin/helm
+            - .
   go_restore_cache:
-    description: "Restore golang cache"
+    description: Restore golang cache
     steps:
-      - restore_cache:
-          key: env-{{ .Revision }}
+      - attach_workspace:
+          # Must be absolute path or relative path from working_directory
+          at: /home/circleci/go
   python_install_requirements:
-    description: "Install requirements in requirements.txt using pip"
+    description: Install requirements in requirements.txt using pip
     steps:
       - run:
           name: Install Python requirements
           command: pip3 install -r tasks/requirements.txt
   alpine_install_git:
-    description: "Install required dependencies"
+    description: Install required dependencies
     steps:
       - run:
           name: Install git
-          command: apk update && apk add git openssh
+          command: apk add --no-cache git openssh
+  ebpf_builder_save_cache:
+    description: Save ebpf-builder tar for later use
+    parameters:
+      path:
+        type: string
+        default: bin/ebpf-builder/ebpf-builder.tar.gz
+        description: ebpf-builder tar location
+    steps:
+      - run: docker save ebpf-builder-amd64 -o <<parameters.path>>
+      - save_cache:
+          key: v1-ebpf-builder-{{ checksum "bin/ebpf-builder/Dockerfile" }}
+          paths:
+            - <<parameters.path>>
+  ebpf_builder_restore_cache:
+    description: Restore ebpf-builder tar from cache
+    parameters:
+      path:
+        type: string
+        default: bin/ebpf-builder/ebpf-builder.tar.gz
+        description: ebpf-builder tar location
+    steps:
+      - restore_cache:
+          keys:
+            - v1-ebpf-builder-{{ checksum "bin/ebpf-builder/Dockerfile" }}
+            # if cache for exact version of docker image is not present, then load any most recent one
+            - v1-ebpf-builder-
+      - run: docker load -i <<parameters.path>> || true
 
 templates:
   # sets the working directory to the project path
   working_directory: &working_directory
-    working_directory: /go/src/github.com/DataDog/chaos-controller
+    working_directory: /home/circleci/go/src/github.com/DataDog/chaos-controller
 
 executors:
-  golang:
+  golang: &golang
     <<: *working_directory
     docker:
       # TODO: Publicize the chaos controller ci image to the datadog docker registry
       #- image: datadog/chaos-controller-runner-circle:<< pipeline.parameters.CURRENT_CI_IMAGE >>
-      - image: golang:1.18-buster
-    environment:
-      KUBEBUILDER_VERSION: 3.1.0
-      KUBEBUILDER_ARCH: amd64
-      GOLANGCI_LINT_VERSION: 1.51.0 ## this value should be the same as in the Makefile
-      HELM_VERSION: 3.5.4
+      - image: cimg/go:1.18
```

Comment:
This is circle ci images, provides default tool installed (like docker) to ease step definition and avoid apt-get/update things